internal/pkg/aws/resourcegroups/resourcegroups.go

```diff
@@ -32,69 +27,63 @@ type tagFilter struct {
 	Values []string
 }
 
-type query struct {
-	ResourceTypeFilters []string
-	TagFilters          []tagFilter
+// Resource contains the ARN and the tags of the resource.
+type Resource struct {
+	Arn  string
+	Tags map[string]string
 }
 
 // New returns a ResourceGroup struct configured against the input session.
 func New(s *session.Session) *ResourceGroups {
 	return &ResourceGroups{
-		client: resourcegroups.New(s),
+		client: resourcegroupstaggingapi.New(s),
 	}
 }
 
-// GetResourcesByTags internally sets the type to TAG_FILTERS_1_0 and generates the query with input resource type and tags.
-func (rg *ResourceGroups) GetResourcesByTags(resourceType string, tags map[string]string) ([]string, error) {
-	var resourceArns []string
-	resourceResp := &resourcegroups.SearchResourcesOutput{}
-
-	query, err := rg.searchResourcesQuery(resourceType, tags)
-	if err != nil {
-		return nil, fmt.Errorf("construct search resource query: %w", err)
+// GetResourcesByTags gets tag set and ARN for the resource with input resource type and tags.
+func (rg *ResourceGroups) GetResourcesByTags(resourceType string, tags map[string]string) ([]*Resource, error) {
+	var resources []*Resource
+	var err error
+	resourceResp := &resourcegroupstaggingapi.GetResourcesOutput{}
+	var tagFilter []*resourcegroupstaggingapi.TagFilter
+	for k, v := range tags {
+		tagFilter = append(tagFilter, &resourcegroupstaggingapi.TagFilter{
+			Key:    aws.String(k),
+			Values: aws.StringSlice([]string{v}),
+		})
 	}
 	for {
-		resourceResp, err = rg.client.SearchResources(&resourcegroups.SearchResourcesInput{
-			NextToken: resourceResp.NextToken,
-			ResourceQuery: &resourcegroups.ResourceQuery{
-				Type:  aws.String(resourceQueryType),
-				Query: aws.String(string(query)),
-			},
+		resourceResp, err = rg.client.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+			PaginationToken:     resourceResp.PaginationToken,
+			ResourceTypeFilters: aws.StringSlice([]string{resourceType}),
+			TagFilters:          tagFilter,
 		})
 		if err != nil {
-			return nil, fmt.Errorf("search resource group with resource type %s: %w", resourceType, err)
+			return nil, fmt.Errorf("get resource: %w", err)
 		}
-		for _, identifier := range resourceResp.ResourceIdentifiers {
-			arn := aws.StringValue(identifier.ResourceArn)
-			resourceArns = append(resourceArns, arn)
+		for _, resourceTagMapping := range resourceResp.ResourceTagMappingList {
+			tags := make(map[string]string)
+			for _, tag := range resourceTagMapping.Tags {
+				if tag.Key == nil {
+					continue
+				}
+				tags[*tag.Key] = aws.StringValue(tag.Value)
```

Comment:
isnt it possible that the value is nil? maybe:
```go
tags[*tag.Key] = ""
if tag.Value != nil {
  tags[*tag.Key] = aws.StringValue(tag.Value)
}
```

Reply:
"StringValue returns the value of the string pointer passed in or if the pointer is nil."

Reply:
lol that's good to know